docs/guides/migrate_to_instance_config_and_disk_resources.md

```diff
@@ -0,0 +1,252 @@
+  id = "49667786,52599108"
+}
+
+import {
+  to = linode_instance_disk.test_imported_disk
+  id = "49667786,98757139"
+}
+
+import {
+  to = linode_instance_disk.test_imported_disk_swap
+  id = "49667786,98757126"
+}
+
+```
+
+generated.tf
+
+```terraform
+# __generated__ by Terraform
+# Please review these resources and move them into your main configuration files.
+
+# __generated__ by Terraform from "49667786,98757139"
+resource "linode_instance_disk" "test_imported_disk" {
+  authorized_keys  = null
+  authorized_users = null
+  filesystem       = "ext4"
+  image            = null
+  label            = "test-disk"
+  linode_id        = 49667786
+  root_pass        = null # sensitive
+  size             = 24576
+  stackscript_data = null # sensitive
+  stackscript_id   = null
+}
+
+# __generated__ by Terraform from "49667786,98757126"
+resource "linode_instance_disk" "test_imported_disk_swap" {
+  authorized_keys  = null
+  authorized_users = null
+  filesystem       = "swap"
+  image            = null
+  label            = "test-swap"
+  linode_id        = 49667786
+  root_pass        = null # sensitive
+  size             = 1024
+  stackscript_data = null # sensitive
+  stackscript_id   = null
+}
+
+# __generated__ by Terraform from "49667786,52599108"
+resource "linode_instance_config" "test_imported_config_1" {
+  booted       = true
+  comments     = null
+  kernel       = "linode/grub2"
+  label        = "test-config-1"
+  linode_id    = 49667786
+  memory_limit = 0
+  root_device  = "/dev/sda"
+  run_level    = "default"
+  virt_mode    = "paravirt"
+  device {
+    device_name = "sda"
+    disk_id     = 98757139
+    volume_id   = 0
+  }
+  device {
+    device_name = "sdb"
+    disk_id     = 98757126
+    volume_id   = 0
+  }
+  helpers {
+    devtmpfs_automount = true
+    distro             = true
+    modules_dep        = true
+    network            = true
+    updatedb_disabled  = true
+  }
+  interface {
+    ipam_address = null
+    label        = null
+    purpose      = "public"
+  }
+  interface {
+    ipam_address = null
+    label        = "test-vlan1"
+    purpose      = "vlan"
+  }
+}
+```
+
+shell output
+
+```
+Terraform has generated configuration and written it to generated.tf. Please review the configuration and edit it as necessary before adding it to version control.
+```
+
+## Apply
+
+Finally, we can run `terraform apply` to put the imported config
+into the states.
+
+Don't forget to double check the sensitive values because TF might not be able to generate them.
+
+```
+Apply complete! Resources: 3 imported, 0 added, 0 changed, 0 destroyed.
+```
+
+## Future
+
+Ideally, we would like to import the resources with ids stored in the states,
+but it's currently not supported by the TF.
+
+I put up an [issue](https://github.com/hashicorp/terraform/issues/33880)
+in their repo for it, and later realized they already implemented it. This feature will be available in TF v1.6
+according to their plan on GitHub.
```

Comment:
I think we can drop the issue reference and just say this feature will be available in `v1.6` 👍